bdns/mocks.go

```diff
@@ -99,6 +99,13 @@ func (mock *MockDNSResolver) LookupCAA(_ context.Context, domain string) ([]*dns
 		fallthrough
 	case "servfail.com":
 		return results, fmt.Errorf("SERVFAIL")
+	case "multi-crit-present.com":
+		record.Flag = 1
+		record.Tag = "issue"
+		record.Value = "symantec.com"
+		results = append(results, &record)
+		record.Value = "letsencrypt.org"
+		results = append(results, &record)
```

Comment:
Hold on a minute. record is of type dns.CAA. You're appending a pointer to it, not a copy of it, twice. In line 107, you change the value of it to letsencrypt.org. So this ends up being just a CAA record for letsencrypt.org, not letsencrypt.org and symantec.com.